server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -503,14 +530,28 @@ public Query rangeQuery(
                         u = FloatPoint.nextDown(u);
                     }
                 }
-                if (isSearchable && hasDocValues) {
+                if (isSearchable) {
                     Query query = FloatPoint.newRangeQuery(field, l, u);
-                    Query dvQuery = SortedNumericDocValuesField.newSlowRangeQuery(
-                        field,
-                        NumericUtils.floatToSortableInt(l),
-                        NumericUtils.floatToSortableInt(u)
+
+                    if (hasDocValues) {
```

Comment:
Can this be moved outside as we are doing the same thing at line 557.

Reply:
We can take the middle ground and add ToDOs or maybe open up refactor backlog issues. 
I empathize that lot of it is legacy code so refactoring can be daunting.